app/src/org/commcare/gis/MapboxLocationPickerActivity.kt

```diff
@@ -57,10 +54,12 @@ class MapboxLocationPickerActivity : BaseMapboxActivity(), CommCareLocationListe
     private var symbolManager: SymbolManager? = null
     private var symbol: Symbol? = null
     private val mapClickListener = MapboxMap.OnMapClickListener { point ->
-        isManualSelectedLocation = true
-        // Add marker.
-        updateMarker(point)
-        viewModel.reverseGeocode(point)
+        if (!Permissions.missingAppPermission(this, LOCATION_PERMISSIONS)) {
```

Comment:
Why do we need the check for location permission for manual selection ?

Reply:
We don't want to respond to map click events if the location permissions haven't been granted. So this checks prevents us from doing that.

Reply:
> We don't want to respond to map click events if the location permissions haven't been granted 

I am not totally sure that's a right statement or if there is a valid reasoning behind that. Location permissions allows  phone to access to the current location of the user and here itself is choosing the location on map. User should be allowed to mark his current location without giving phone/CC access to his location and not doing so wouldn't be a good citizen Android policy as we are forcing user to give CC access to his location even when he doesn't want/need to.

Reply:
>Location permissions allows phone to access to the current location of the user and here itself is choosing the location on map.

Without the location permission, the map load event won't take the user to his current location. It would rather be a whole map and then user will have to zoom in and navigate all the way to their current location. Is that something we should support? 
Wouldn't that cause issues if a user clicks on any weird location and simply confirms it(without verifying that it's a correct location), and then later on project would complain that the captured locations aren't correct.

Reply:
That's a good point. Do we know what's the behaviour for normal map activity. Probably best to be consistent with that here.

Reply:
I checked the normal map activity. It finishes if a user denies location permission. 
I think this is gonna be tricky to do with mapbox activity because the `KujakuMapView` is handling the location permissions internally. So if we use our own permissions check outside, then there will be 2 permission popups back to back which will be a weird user experience.

Reply:
sounds good, though we should atleast show a message to the user here if there is no permisson given otherwise user will keep clicking the map without any UI feedback

Reply:
Yup, that sounds good to me. Will display a toast.